bpf/lib/nodeport.h

```diff
@@ -1475,7 +1478,7 @@ handle_dsr_v4(struct __ctx_buff *ctx, struct iphdr *ip4, bool *dsr)
 	/* Check whether IPv4 header contains a 64-bit option (IPv4 header
 	 * w/o option (5 x 32-bit words) + the DSR option (2 x 32-bit words)).
 	 */
-	if (ip4->ihl == 0x7) {
+	if (ip4->ihl >= 0x7) {
```

Comment:
If we allow `ihl > 7` it would mean 2 or more IP options, not just the one. The body of this statement still expects the DSR option to be at the top, but is that always the case if we have 2 or more options?

Reply:
yep, see commit description:

> Note that we always insert the DSR option at the front, so it's sufficient
> to parse the first option.

If something else inserts additional IP options mid-route, we're in trouble of course. But hey, one step at a time :shrug:.

Reply:
Alright, that is fair then. Didn't see the commit message. Perhaps we should, at some point add a `find_dsr_v6` equivalent, but I agree that that doesn't seem needed in this PR.

Reply:
> Alright, that is fair then. Didn't see the commit message. Perhaps we should, at some point add a `find_dsr_v6` equivalent, but I agree that that doesn't seem needed in this PR.

Agree, would be a good follow-on improvement (even just to get things on par). There's the added complication of adding more code into `nodeport_lb4()` of course :sweat_smile:.